docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -460,6 +468,45 @@ secret URI.
         :start-at: config:
         :end-before: containers:
 
+.. admonition:: Best practice
+    :class: hint
+
+    Make the user experience of the operator as smooth as possible. The
+    application should be deployable without providing any further configuration
+    options: ``juju deploy foo`` is preferable to
+    ``juju deploy foo --config required-option=value``.
+
+.. admonition:: Best practice
+    :class: hint
+
+    Just like Juju, a charm is an opinionated tool. Configure the application
+    with the best defaults, and only expose application configuration options
+    when necessary.
```

Comment:
Do you think we could combine these points? They feel almost the same to me. I prefer the copy in the second paragraph, at any rate.